article/app/model/dotcomponents/DotcomponentsDataModel.scala

```diff
@@ -304,13 +255,31 @@ object DotcomponentsDataModel {
         main.getOrElse(Configuration.images.fallbackLogo)
       }
 
-      NewsArticle(
-        `@id` = article.metadata.webUrl,
-        images = Seq(ImgSrc(mainImageURL, Item1200)),
-        author = article.tags.contributors.mkString(", "),
-        datePublished = article.trail.webPublicationDate.toString(),
-        dateModified = article.fields.lastModified.toString(),
-        headline = article.trail.headline,
+      val authors = article.tags.contributors.map(contributor => {
+        Person(
+          name = contributor.name,
+          sameAs = contributor.metadata.webUrl,
+        )
+      })
+
+      List(
+        NewsArticle(
+          `@id` = Configuration.amp.baseUrl + article.metadata.id,
```

Comment:
It's not clear (to me..) from reading the docs what `@id` in the newsArticle context is used for, however - is this purely `AMP` or is this structured data that will be used for the main article pages in DC too? Feel free to just reply with a link to a page on schema.org or whatever that explains!

Reply:
good point! This is AMP specific. I did this to mirror current behaviour. If we use the regular URL here then the NewsArticle and WebPage structured data fields get merged (as same ID), which I think from a SD sense is fine, but I was wary of differing from PROD here as I'd prefer to get the same behaviour first and then improve. Having said that, I haven't fully mirrored prod here.